common/core/web/input-processor/tests/cases/inputProcessor.js

```diff
@@ -42,4 +42,114 @@ describe('InputProcessor', function() {
       assert.isTrue(core.languageProcessor.mayPredict);
     });
   });
+
+  describe('efficiency tests', function() {
+    let testDistribution = [];
+    var keyboard;
+    let device = {
+      formFactor: 'phone',
+      OS: 'ios',
+      browser: 'safari'
+    }
+
+    // Easy peasy long context:  use the input processor's full source!
+    let coreSourceCode = fs.readFileSync('dist/index.js', 'utf-8');
+
+    // At the time this test block was written...  810485 chars.
+    // Let's force it to the same order of magnitude, even if the codebase grows.
+    if(coreSourceCode._kmwLength() > 1000000) {
+      coreSourceCode = coreSourceCode._kmwSubstring(0, 1000000);
```

Comment:
```suggestion
    if(coreSourceCode.length() > 1000000) {
      coreSourceCode = coreSourceCode.substring(0, 1000000);
```

There's really no reason to use `kmw` forms here is there?

Reply:
Eh, that's fair.  I used it mostly for cross-unit-test consistency.  True, KMW's source code doesn't actually have SMP characters.

Reply:
`length` is a property, not a method, though...